http_proxy_test.go

```diff
@@ -509,40 +505,42 @@ func basicServer(maxConns uint64, idleTimeout time.Duration) *server.Server {
 	return srv
 }
 
-func setupNewHTTPServer(maxConns uint64, idleTimeout time.Duration) (*server.Server, error) {
+func setupNewHTTPServer(maxConns uint64, idleTimeout time.Duration) (string, error) {
 	s := basicServer(maxConns, idleTimeout)
 	var err error
 	ready := make(chan string)
 	wait := func(addr string) {
+		log.Debugf("Started HTTP proxy server at %s", addr)
 		ready <- addr
 	}
 	go func(err *error) {
-		if *err = s.ServeHTTP("localhost:0", wait); err != nil {
+		if *err = s.ListenAndServeHTTP("localhost:0", wait); err != nil {
```

Comment:
What would happen in case that `ListenAndServeHTTP` fails? looks like `readyCb` is only called on success, so if it fails, would `<-ready` continue blocking?